### PR TITLE
Navigation update

### PR DESCRIPTION
**Pull request by MustafaMadkour**
# Navigation Links API
## _Description_
 API to create navigation bar links and retrieve them from a mongodb database.
this is the first update to be compatible with the JSON server db. 

### Stack:
We used Nestjs - in typescript -and Mongodb as a databases.

### Request Body:
```````
{
    "title":  "link-title",
    "url":  "link-url",
}
```````

### Requests:  
we have these requests available:
- GET `localhost:3001/links/:id` to get a navigation link by its id
- GET `localhost:3001/links` to get all navigation links

### Tasks to complete:
- [x] NavigationLinks-Endpoints to db
- [x] Error Handler
- [x] Error Logger

### CODE REVIEW

File: api/src/navLinks/navLinks.controller.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  findAll(): Promise<Links[]> {
+    return this.linksService.findAll();
+  }
+
+  @Get(':id')
+  async findOne(@Param('id') id): Promise<Links> {
+    return this.linksService.findOne(id);
+  }
+
+  @Post()
```
Comment (Nilegfx) [inline api/src/navLinks/navLinks.controller.ts]: remove write apis. 

we should add the links manually for now, once we implemented the auth, we can start the write apis

File: api/src/models/estimate.schema.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import * as mongoose from 'mongoose';
```
Comment (Nilegfx) [inline api/src/models/estimate.schema.ts]: remove this and other schemas from this PR and create a separate PR for the rest of schemas.

File: api/src/models/navLinks.schema.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import * as mongoose from 'mongoose';
```
Comment (Nilegfx) [inline api/src/models/navLinks.schema.ts]: name this file consistently, the resource called `links`, then lal files, modules or related code should be links not `navLinks`

File: api/src/navLinks/dto/create-navLinks.dto.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+
+export class CreateLinksDto {
```
Comment (Nilegfx) [inline api/src/navLinks/dto/create-navLinks.dto.ts]: why do you name this `create`LinkeDto? can't we name it linksDto? I am asking real question not requesting you to change it.

File: api/src/navLinks/navLinks.controller.spec.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { Test, TestingModule } from '@nestjs/testing';
```
Comment (Nilegfx) [inline api/src/navLinks/navLinks.controller.spec.ts]: name this file links not navlinks

File: api/src/navLinks/navLinks.module.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import { Module } from '@nestjs/common';
```
Comment (Nilegfx) [inline api/src/navLinks/navLinks.module.ts]: name it links not navLinks

File: api/src/navLinks/navLinks.service.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@nestjs/common';
```
Comment (Nilegfx) [inline api/src/navLinks/navLinks.service.ts]: name it links not navLinks

File: api/src/types/navLinks.ts [hunk 8]
```diff
@@ -0,0 +1,6 @@
+import { Document } from 'mongoose';
```
Comment (Nilegfx) [inline api/src/types/navLinks.ts]: name this li ks not navLinks

**Review by Nilegfx — CHANGES_REQUESTED** [review]
please
* name modules, functions and file names consistently 
* remove write apis
* remove unrelated schemas and create separate PR for them

File: api/src/navLinks/dto/create-navLinks.dto.ts (hunk 4 above)
Comment (MustafaMadkour) [reply api/src/navLinks/dto/create-navLinks.dto.ts]: cuz i understood that the dto is to create a doc in a specific type ?
but as i'm learning r8 nw that thought was wrong.

**Comment by MustafaMadkour** [comment]
all changes requested will be considered in the next pr

File: api/src/navLinks/navLinks.controller.ts (hunk 1 above)
Comment (MustafaMadkour) [reply api/src/navLinks/navLinks.controller.ts]: ok that was discussed was Abdullah after submitting this pr 
changes will be done.

File: api/src/navLinks/dto/create-navLinks.dto.ts (hunk 4 above)
Comment (MustafaMadkour) [reply api/src/navLinks/dto/create-navLinks.dto.ts]: DTo gives us the upper hand to decide what data-schema can the client-side and the server-side can exchange in a single request/response cycle

**Review by Nilegfx — APPROVED** [review]
LGTM now.

thanks for your hard work, dedication and standing by your goals and deliverables. I like this behavior so much, Mustafa.

Good job!

**Pull request merged**